backends/ebpf/run-ebpf-test.py

```diff
@@ -196,7 +196,10 @@ def run_test(options, argv):
                 exp_stderr = f.read().strip()
             # Output will have full path which differs
             out = out.replace(options.p4filename, basename).strip()
-            # Output will have one extra line with make error, trim that
+            # Output may start with annoying "jobserver unavailable" make warning, trim that
+            if out[0:5] == "make[":
```

Comment:
This looks strange to me, what sort of failure produces this error? Can we fix the source of the problem?

Reply:
I tried quite hard to fix the source but it seems like there is no way to force CMake to do the right thing, which is why we end up with this kludge.

The issue is that, when running `make -j4 check` (notice the `-j`) argument, make emits a "jobserver unavailable" warning which thus causes the output to not match.

https://stackoverflow.com/questions/33171335/jobserver-unavailable-when-building-external-projects-with-cmake

Reply:
If I understand this right, CI might fail because in the error output there is a spurious `jobserver unavailable` in the output, which mismatches with the golden tests?

And we get this spurious output because we compile the generated C code with `make -jX`? Can we add this as a comment to clarify?

Reply:
We do use `-jX` (https://github.com/p4lang/p4c/blob/main/.github/workflows/ci-test.yml#L21) but CI nodes only have 2 cores iirc. 

Even so, the output that is checked here comes from the eBPF test and make, which is triggered in functions like this one:
https://github.com/p4lang/p4c/blob/main/backends/ebpf/targets/target.py#L63
https://github.com/p4lang/p4c/blob/main/backends/ebpf/targets/kernel_target.py#L43

Reply:
I was talking about running `make -j4 check` to run the test suite -- it seems like the CI calls `ctest` directly, which may be why we're seeing different behaviours (because we no longer have `make` as a parent process).

The make calls in the python target files are the ones which emit the warning, yes.

Reply:
Looks like `make -j4 check` causes subprocesses that use make to also run in parallel. Might be worthwhile to simply fix the make calls of the script to be single-threaded. The parallelism is not intended.

Reply:
What you're saying is a simple `make check` runs in parallel already, so running `make -j4 check` is invalid? I guess we could just drop this change then.

Reply:
I did try adding `-j1` there and it didn't suppress the warning.

Honestly I'm now of the opinion that it's on me for running `make check` with `-j4` and I should just stop doing that. We have 3 options:

1. Keep this patch as it is (doesn't hurt as far as I can tell) but add some more comments to clarify the issue.
2. Try to fix the root cause in the CMake-Make-Python-Make chain.
3. Drop this patch since it doesn't fix anything in CI anyway.

Reply:
Naturally, I prefer 2 (I believe we need to add `-j1` to all make calls in the script?) but I am also okay with option 1 as long as we clearly document the cause, as you also pointed out.

Reply:
I can't figure it out - just adding `-j1` does not solve it. In the meanwhile I've improved the comment here.